### DISCUSSION TITLE
How do I publish .dll packages into github packages.

### DISCUSSION THREAD

**Discussion by ghost** [community/community#78929] [Packages]
### Select Topic Area

Question

### Body

How do I publish .dll packages into github packages.

**Comment by erickdc7** [comment]
To publish .dll packages into GitHub Packages, you need to use NuGet, which is a package manager for .NET Framework and .NET Core. NuGet allows you to create and publish packages that contain .dll files and other resources.

Here are the general steps to publish .dll packages into GitHub Packages using NuGet:

- **Create a NuGet package**. You need to create a .nuspec file that describes the metadata and contents of your package, such as the name, version, dependencies, and files. You can use the `nuget spec` command to generate a template .nuspec file, and then edit it according to your needs. You also need to organize your .dll files and other resources into a folder structure that follows the NuGet conventions.
- **Pack the NuGet package**. You need to use the `nuget pack` command to create a .nupkg file from your .nuspec file and your folder structure. This .nupkg file is the package that you will publish to GitHub Packages.
- **Push the NuGet package**. You need to use the `nuget push` command to upload your .nupkg file to GitHub Packages. You also need to authenticate to GitHub Packages using a personal access token (classic) with the appropriate scopes, and specify the GitHub Packages URL as the source for your package.

For more detailed instructions and examples, you can refer to this GitHub gist that shows how to publish .dll packages into GitHub Packages using NuGet.

I hope this helps!
